herd/AArch64Sem.ml

```diff
@@ -1020,13 +1020,19 @@ module Make
               fire_spurious_af dir a m) >>= M.ignore >>= B.next1T)
             mphy
             mfault in
-        M.delay_kont "6" ma
-          (if pte2 then maccess
-           else
+        M.delay_kont "6" ma (
+          if pte2 then maccess
+          else
              fun a ma ->
              match Act.access_of_location_std (A.Location_global a) with
-             | Access.VIR|Access.PTE -> maccess a ma
-             | ac -> mop ac ma >>= M.ignore >>= B.next1T)
+             | Access.VIR|Access.PTE ->
+                if A.V.is_instrloc a then
+                 (* Ignore VMSA when loading/storing to code locations. *)
+                 (mop Access.VIR ma >>= M.ignore >>= B.next1T)
```

Comment:
Is there any reason we change the access type here? See also my suggestion below.

Reply:
No specific reason. It occurred to me an access to an instruction location should not have the `Access.PTE` type, but there is no reason to enforce it here.